view/business-process-revision.js

```diff
@@ -168,64 +168,87 @@ exports._pauseButton = function (/*options*/) {
 };
 
 exports._revisionContainer = function () {
-	var paymentUploads = getUploads(this.processingStep.paymentReceiptUploads, this.appName);
-	section({ class: 'section-tab-nav' },
-		a({
+	var requirementUploads    = this.processingStep.requirementUploads
+	  , paymentReceiptUploads = this.processingStep.paymentReceiptUploads
+	  , uploads               = getUploads(requirementUploads, this.appName)
+	  , paymentUploads        = getUploads(paymentReceiptUploads, this.appName);
+
+	var paymentUploadsTabNumber = resolve(uploads, '_length').map(function (uploadsLength) {
```

Comment:
Let's keep `paymentUploadsTabNumber` as an observable which stores a number. We should have observable for each tab number. And they should actually store numbers. One should calculate it's number as previous + 1. Then, once you pass those observables to i18n placeholders use `map` to transform the number value to string.